### ISSUE TITLE
JS.patch/1 fails to patch to route under scope which includes a host

### ISSUE THREAD

**Issue by PolyCement** [phoenixframework/phoenix_live_view#2665]
### Environment

* Elixir version (elixir -v): 1.14.0
* Phoenix version (mix deps): 1.7.2
* Phoenix LiveView version (mix deps): 0.18.18
* Operating system: Linux 6.3.4-arch1-1
* Browsers you attempted to reproduce this bug on (the more the merrier): Firefox, Chromium
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: Directory doesn't exist

### Actual behavior

When using `JS.patch/1` to patch to a route under a scope which includes a host, I get the following error:

```
[error] GenServer #PID<0.631.0> terminating
** (ArgumentError) cannot invoke handle_params nor live_redirect/live_patch to "/users" because it isn't defined in TestZone2Web.Router
    (phoenix_live_view 0.18.18) lib/phoenix_live_view/route.ex:43: Phoenix.LiveView.Route.live_link_info!/3
    (phoenix_live_view 0.18.18) lib/phoenix_live_view/channel.ex:118: Phoenix.LiveView.Channel.handle_info/2
    (stdlib 4.3) gen_server.erl:1123: :gen_server.try_dispatch/4
    (stdlib 4.3) gen_server.erl:1200: :gen_server.handle_msg/6
    (stdlib 4.3) proc_lib.erl:250: :proc_lib.wake_up/3
Last message: %Phoenix.Socket.Message{topic: "lv:phx-F2XEnaDroUXCkgBk", event: "live_patch", payload: %{"url" => "/users"}, ref: "42", join_ref: "21"}
State: %{components: {%{2 => {TestZone2Web.UserLive.FormComponent, 2, %{__changed__: %{}, action: :edit, flash: %{}, form: %Phoenix.HTML.Form{source: #Ecto.Changeset<action: nil, changes: %{}, errors: [], data: #TestZone2.Accounts.User<>, valid?: true>, impl: Phoenix.HTML.FormData.Ecto.Changeset, id: "user", name: "user", data: %TestZone2.Accounts.User{__meta__: #Ecto.Schema.Metadata<:loaded, "users">, id: 2, age: 30, name: "Test User", inserted_at: ~N[2023-06-05 13:02:00], updated_at: ~N[2023-06-05 13:02:00]}, hidden: [id: 2], params: %{}, errors: [], options: [method: "put"], index: nil, action: nil}, id: 2, myself: %Phoenix.LiveComponent.CID{cid: 2}, patch: "/users", title: "Edit User", user: %TestZone2.Accounts.User{__meta__: #Ecto.Schema.Metadata<:loaded, "users">, id: 2, age: 30, name: "Test User", inserted_at: ~N[2023-06-05 13:02:00], updated_at: ~N[2023-06-05 13:02:00]}}, %{__changed__: %{}, lifecycle: %Phoenix.LiveView.Lifecycle{handle_event: [], handle_info: [], handle_params: [], mount: [], after_render: []}, root_view: TestZone2Web.UserLive.Index}, {31608148928896753784572504986217761096, %{0 => {225870775518966524544372673426077242922, %{1 => {275660793971508029625723670889715429128, %{}}, 2 => {15508784672739301999445006821212520444, %{0 => {89864552659920068383128005838426367669, %{}}}}}}, 1 => {238806784837768290486236055472832185344, %{0 => {118970292807654719229453556541006727223, %{3 => {229926359671685626713800839236463170965, %{0 => {33639875295649801095848358102975471311, %{0 => {170257016188728605361315375560329013142, %{1 => {75443377559193289484622560952526872310, %{1 => {244639731801124985314772032709064439773, %{}}}}}}, 1 => {170257016188728605361315375560329013142, %{1 => {75443377559193289484622560952526872310, %{1 => {244639731801124985314772032709064439773, %{}}}}}}}}, 1 => 119022389494709290857508506158882350949}}}}}}}}}}, %{TestZone2Web.UserLive.FormComponent => %{2 => 2}}, 3}, join_ref: "21", serializer: Phoenix.Socket.V2.JSONSerializer, socket: #Phoenix.LiveView.Socket<id: "phx-F2XEnaDroUXCkgBk", endpoint: TestZone2Web.Endpoint, view: TestZone2Web.UserLive.Index, parent_pid: nil, root_pid: #PID<0.631.0>, router: TestZone2Web.Router, assigns: %{__changed__: %{}, flash: %{}, live_action: :edit, page_title: "Edit User", streams: %{__changed__: MapSet.new([]), users: %Phoenix.LiveView.LiveStream{name: :users, dom_id: #Function<3.113057034/1 in Phoenix.LiveView.LiveStream.new/3>, inserts: [], deletes: []}}, user: %TestZone2.Accounts.User{__meta__: #Ecto.Schema.Metadata<:loaded, "users">, id: 2, age: 30, name: "Test User", inserted_at: ~N[2023-06-05 13:02:00], updated_at: ~N[2023-06-05 13:02:00]}}, transport_pid: #PID<0.615.0>, ...>, topic: "lv:phx-F2XEnaDroUXCkgBk", upload_names: %{}, upload_pids: %{}}
```

In this example, the scope in `TestZone2Web.Router` looks like this:

```
  scope "/", TestZone2Web, host: "example." do
    pipe_through :browser

    get "/test", PageController, :home
    
    live "/users", UserLive.Index, :index
    live "/users/new", UserLive.Index, :new
    live "/users/:id/edit", UserLive.Index, :edit

    live "/users/:id", UserLive.Show, :show
    live "/users/:id/show/edit", UserLive.Show, :edit
  end
```

Repo with example project [here](https://github.com/PolyCement/patch-bug-demo). Steps to reproduce:
1. Navigate to [http://example.localhost:4000/users](http://example.localhost:4000/users)
2. Add a user
3. Click "Edit" next to new user's name
4. Close out the modal (either with the close button or by clicking outside the modal)

I've checked that the route is valid using `Phoenix.Router.route_info/4`, so I'm assuming the host isn't being passed along properly.

### Expected behavior

The page should correctly patch to `http://example.localhost:4000/users`, like how it patches to `http://localhost:4000/users` if I remove the host from the scope.

**Comment by CoderDennis** [comment]
I can reproduce the error and am investigating a fix.

**Comment by CoderDennis** [comment]
@PolyCement if you change `on_cancel={JS.patch(~p"/users")}` to `on_cancel={JS.navigate(~p"/users")}` for your modal it works without the error.

**Comment by PolyCement** [comment]
I'd noticed that too, but it's not an ideal workaround, so I did some poking around in the code to figure out what was going on. It looks like the difference between the two is that `historyRedirect` converts relative paths to full paths, while `pushHistoryPatch` doesn't:

https://github.com/phoenixframework/phoenix_live_view/blob/b2ff167634adc6046cac9f10071d22ad122dafb6/assets/js/phoenix_live_view/live_socket.js#L770-L773

If the code is working like I think it is, then the error occurs when the relative path is passed to `pushLinkPatch` - the path doesn't include a host, so Phoenix can't find it in the router. Copying the above 4 lines into `pushHistoryPatch` ensures the host is included, fixing the issue.

I've submitted a PR here: #2683
